src/main/java/com/sarality/sync/APIResponseHandler.java

```diff
@@ -0,0 +1,13 @@
+package com.sarality.sync;
+
+/**
+ * Interface for handlers for API Responses
+ *
+ * @author Satya@ (Satya Puniani)
+ */
+
+public interface APIResponseHandler<T, R> {
+
+  public T process(T sourceData, R responseData);
```

Comment:
Again might be nice to document what T and R is. It is clear that R is the response, but not sure what T is. Is it the Data that is being synced or the SyncRequest object.